recording-daemon/packet.c

```diff
@@ -177,9 +176,26 @@ static ssrc_t *ssrc_get(stream_t *stream, unsigned long ssrc) {
 	dbg("Init for SSRC %s%lx%s of stream #%lu", FMT_M(ret->ssrc), stream->id);
 
 	if (mf->recording_on && !ret->output && output_single) {
-		char buf[16];
-		snprintf(buf, sizeof(buf), "%08lx", ssrc);
-		ret->output = output_new(output_dir, mf->parent, buf);
+		ilog(LOG_INFO, "Metadata %s and output %s", mf->metadata, mf->output_dest);
+		if (mf->output_dest) {
+			char path[256];
+			strncpy(path, mf->output_dest, sizeof(path));
+			char *sep = strrchr(path, '/');
+			if (sep) {
+				char *filename = sep + 1;
+				*sep = 0;
+				ret->output = output_new(path, mf->parent, filename);
```

Comment:
I'm a bit confused about this. The 3rd argument to output_new() is supposed to be the "type" of the stream (format `%t`), which in case of single outputs is supposed to be the SSRC (or some other unique identifier). This seems to change the `%t` format to a completely different purpose. Is this intended? AFAICT the `output_dest` should just replace the output format string and/or the path used to create the full filename.